### PR TITLE
New sample: Line of sight (location)

### PR DESCRIPTION
**Pull request by ghost**
This sample demonstrates how to interactively place a line of sight between two locations.

### CODE REVIEW

**Review by nixta — CHANGES_REQUESTED** [review]
This is a bit static. Once the observer and target have been set, there is no interactive way to modify the target again.

The demo should really allow the user to drag the target position around and see how the Line of Sight analysis updates in realtime.

This will impact how the instructions UI is set up. @gdhakal I have pulled together a suggestion for how this could work. I'll get with you to see what you think.

**Review by nixta — APPROVED** [review]

File: arcgis-ios-sdk-samples/Analysis/Line of sight (location)/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+![](image1.png)
+
+## How it works
+
+`AGSLocationLineOfSight` analysis provides an initializer called `init(observerLocation:targetLocation:)` that takes observer and target locations.
+
+Once the `AGSLocationLineOfSight` is created, it is added to a collection of analysis overlays in the `AGSSceneView`. The analysis overlays are used to render the results of visual analysis on the scene view.
+
+The sample uses the `geoView(_:didTapAtScreenPoint:mapPoint:)` method on `AGSGeoViewTouchDelegate` to get the tapped point and sets the `observerLocation` on the `AGSLocationLineOfSight`. The `targetLocation` on the `AGSLocationLineOfSight` is updated on `geoView(_:didLongPressAtScreenPoint:mapPoint:)` when user performs long-pressed gesture at a specified location.
```
Comment (nixta) [inline arcgis-ios-sdk-samples/Analysis/Line of sight (location)/README.md]: Oops. I should have included this in my PR. Could we update this text to mention how we use `didMoveLongPress` as well as `didLongPress`?

**Review by nixta — CHANGES_REQUESTED** [review]
Sorry. I should have thought to update the README.md when I submitted my PR of changes. Could you update it to cover the more interactive approach please?

**Comment by gdhakal** [comment]
@nixta Made changes to the sample. Please take a look. 

- Changed viewpoint with new camera position. The imagery displays well in iPhone 8.
- Changed screenshot `image1.png` as viewpoint is modified in the sample. 
- Documented the added map interaction in README. 
- Removed `self` reference in properties to match with the rest of the code (for consistency purpose).

**Review by nixta — APPROVED** [review]

**Comment by dg0yal** [comment]
@gdhakal Code looks good.  Not sure the cyan and magenta colors work for me. I'd much prefer it default to green and red which better conveys what is visible and what isn't. Plus it'll match the other viewshed samples which use a similar color scheme

**Comment by gdhakal** [comment]
@dg0yal The cyan and magenta color is used per sample design documentation.

**Comment by dg0yal** [comment]
Yeah, i don't think that's the best design decision. The cyan color is commonly used for selecting things, so it appears that part of the line is selected, instead of conveying that that part is visible.

**Comment by gdhakal** [comment]
@dg0yal The samples now uses default colors to represent visible and obstructed segments. The screenshot and README files are modified accordingly. Please take a look at it.

**Comment by nixta** [comment]
@gdhakal Please take a look at my latest pushed changes.

**Comment by gdhakal** [comment]
@nixta Thanks Nick. It looks great :shipit:

**Comment by nixta** [comment]
Just did the same, @gdhakal. You triggered my OCD.

**Pull request merged**